### DISCUSSION TITLE
Location fuzzing

### DISCUSSION THREAD

**Discussion by ProducerMatt** [meshtastic/firmware#3130] [Ideas]
The ham radio digital protocol "APRS" has a location fuzzing feature, which randomly displaces your location from 0 to a maximum distance you set. This allows you to, for example, make others aware of your repeater's presence but not actually reveal your house.

If this feature was present in Meshtastic, I would have no problem submitting my repeater's location to public maps. However it is so accurate as to effectively dox me, so I'm absolutely not willing to do that.

I don't know if you can send location broadcasts on Secondary talk groups. If you can, something else that would be useful would be the ability to enable/disable fuzzing per channel, so I could broadcast my location precisely on private channels but fuzzed when on channels for general socializing.

**Comment by HiroProt** [comment]
I like that idea, but also have an additional idea/request: a user configurable list of lat/lng/radius entries in which position updates will NOT be broadcast, so that I can leave GPS on for my mobile nodes and potentially have full precision while on the go, but don't have to worry about it revealing sensitive locations like home/work/others precisely.

**Reply by garthvh** [reply]
Both of these features are already implemented

**Reply by HiroProt** [reply]
Ha! So fast :) Thanks @garthvh

**Reply by HiroProt** [reply]
@garthvh I found the precision fuzzing, but am not seeing the list of privacy zones support. Is there a specific PR / commit that added that?

**Comment by ProducerMatt** [comment]
If I understand this feature addition it fulfills the request: https://github.com/meshtastic/firmware/pull/3263

**Comment by garthvh** [comment]
There are no "zones", that would be a way bigger UX lift, you can set precision per channel.

**Reply by HiroProt** [reply]
Ah, got it. What I'm asking for is less of a precision thing, and more about areas that would essentially stop position updates from going out. I understand that it would be a bit more tricky UX-wise, but I may take a stab at it myself if it's not already in the works. Happy to contribute :)
